### PR TITLE
[staging-next] pythonPackages fixes

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
some package fixes

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jonringer** [comment]
I also had tbm-utils, audio-metadata, and one other, but you already got to them :)

**Comment by jonringer** [comment]
I'm going to do a few more package fixes if you don't mind. Don't merge for a little while :)

File: pkgs/development/python-modules/datadog/default.nix [hunk 1]
```diff
@@ -17,9 +15,12 @@ buildPythonPackage rec {
     find . -name '*.pyc' -exec rm {} \;
   '';
 
-  propagatedBuildInputs = [ decorator requests simplejson ];
+  propagatedBuildInputs = [ decorator requests simplejson pillow ];
 
-  checkInputs = [ nose mock pillow ];
+  checkInputs = [ nose mock pytest ];
+  checkPhase = ''
+    pytest tests/unit
```
Comment (FRidh) [inline pkgs/development/python-modules/datadog/default.nix]: ${python.interpreter}

Comment (jonringer) [reply pkgs/development/python-modules/datadog/default.nix]: ? I'm calling pytest, not python

Comment (FRidh) [reply pkgs/development/python-modules/datadog/default.nix]: ahh, I need a break :stuck_out_tongue:

Comment (jonringer) [reply pkgs/development/python-modules/datadog/default.nix]: that's probably fair :)

**Comment by prusnak** [comment]
Pyro4 build fails with the following for me:

```
$ nix-shell -I nixpkgs=. -p python3Packages.Pyro4

...

=================================== FAILURES ===================================
__________________________ TestSSL.testContextAndSock __________________________

self = <PyroTests.test_socket.TestSSL testMethod=testContextAndSock>

    def testContextAndSock(self):
        cert_dir = "../../certs"
        if not os.path.isdir(cert_dir):
            cert_dir = "../certs"
            if not os.path.isdir(cert_dir):
>               self.fail("cannot locate test certs directory")
E               AssertionError: cannot locate test certs directory

tests/PyroTests/test_socket.py:568: AssertionError
=============================== warnings summary ===============================
tests/PyroTests/test_socket.py:375
  /build/Pyro4-4.75/tests/PyroTests/test_socket.py:375: PytestCollectionWarning: cannot collect test class 'TestDaemon' because it has a __init__ constructor (from: tests/PyroTests/test_socket.py)
    class TestDaemon(Daemon):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
= 1 failed, 602 passed, 20 skipped, 6 deselected, 1 warnings in 77.65s (0:01:17) =
```

**Review by prusnak — CHANGES_REQUESTED** [review]
Pyro4 build fails with the following for me:

``` python
$ nix-shell -I nixpkgs=. -p python3Packages.Pyro4

...
...
...

=================================== FAILURES ===================================
__________________________ TestSSL.testContextAndSock __________________________

self = <PyroTests.test_socket.TestSSL testMethod=testContextAndSock>

    def testContextAndSock(self):
        cert_dir = "../../certs"
        if not os.path.isdir(cert_dir):
            cert_dir = "../certs"
            if not os.path.isdir(cert_dir):
>               self.fail("cannot locate test certs directory")
E               AssertionError: cannot locate test certs directory

tests/PyroTests/test_socket.py:568: AssertionError
=============================== warnings summary ===============================
tests/PyroTests/test_socket.py:375
  /build/Pyro4-4.75/tests/PyroTests/test_socket.py:375: PytestCollectionWarning: cannot collect test class 'TestDaemon' because it has a __init__ constructor (from: tests/PyroTests/test_socket.py)
    class TestDaemon(Daemon):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
= 1 failed, 602 passed, 20 skipped, 6 deselected, 1 warnings in 77.65s (0:01:17) =
```

**Comment by jonringer** [comment]
are you on darwin?

**Comment by prusnak** [comment]
@jonringer no:

``
$ nix-info 
system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.3.1, channels(user): "", channels(root): "nixos-20.03pre198214.4cd2cb43fb3, nixos-hardware", nixpkgs: /nix/var/nix/profiles/per-user/root/channels/nixos
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.Pyro4 python37Packages.Pyro4 python38Packages.Pyro4

**Comment by jonringer** [comment]
@prusnak do you have an overlay present? ofborg was able to build on aarch-linux

**Comment by prusnak** [comment]
hmm no ...

**Comment by jonringer** [comment]
seeing as these builds are broken on staging-next, I'm going to merge, worse case, their still broken for some.

**Pull request merged**